ratis-server/src/main/java/org/apache/ratis/server/leader/LogAppenderDaemon.java

```diff
@@ -44,11 +46,17 @@ class LogAppenderDaemon {
 
   private final LogAppender logAppender;
 
-  LogAppenderDaemon(LogAppender logAppender) {
+  LogAppenderDaemon(LogAppender logAppender, RaftServer.Division server) {
```

Comment:
I have to use `RaftServer.Division` here and cast to `RaftServerImpl` internally because the `server` ref is passed in from `GrpcFactory/ServerFactory` all the way down and `ServerFactory` cannot import `RaftServerImpl`(circular dependency).

Reply:
Is it possible to pass in a `ThreadGroup` here instead of passing in a `RaftServer.Division`

Reply:
That is possible but I will need to pass in the `ThreadGroup` from `RaftServerImpl` [here](https://github.com/apache/ratis/blob/8fbe434efad64436e5505f5a03a4ef048a425a0a/ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java#L245) and essentially modify the signature on [this interface](https://github.com/apache/ratis/blob/8fbe434efad64436e5505f5a03a4ef048a425a0a/ratis-server-api/src/main/java/org/apache/ratis/server/ServerFactory.java#L37). 

So which one do you prefer @codings-dan :
Option 1: Update the interface `ServerFactory.newLogAppender()` to pass in `ThreadGroup`. If this interface is not public-facing then I'll just do that.

Option 2: Do the cast here as the code currently shows.

Reply:
Can we add a `getThreadGroup` method declaration to the `RaftServer` interface, and then this constructor can be directly changed to
```
Daemon.Builder builder = Daemon.newBuilder().setName(name).setRunnable(this::run)
        .setThreadGroup(logAppender.getServer().getThreadGroup());
```

Do you think this is okay?

Reply:
@codings-dan Sounds good to me. I think a getter on the `RaftServer` interface would be totally fine and non-distruptive, with a `default` implementation too.

Reply:
Yean, maybe a default implementation needs to be added as well.